core/src/test/java/com/radixdlt/genesis/olympia/converter/OlympiaToBabylonGenesisConverterResourceSupplyTest.java

```diff
@@ -0,0 +1,186 @@
+            HashCode.fromBytes(ECKeyPair.generateNew().getPublicKey().getCompressedBytes()));
+    final var acc2 =
+        new OlympiaStateIR.Account(
+            HashCode.fromBytes(ECKeyPair.generateNew().getPublicKey().getCompressedBytes()));
+    final var balances =
+        ImmutableList.of(
+            new OlympiaStateIR.AccountBalance(0, 1, BigInteger.valueOf(10000)),
+            new OlympiaStateIR.AccountBalance(0, 1, BigInteger.valueOf(2001)));
+    final var olympiaState =
+        new OlympiaStateIR(
+            ImmutableList.of(),
+            ImmutableList.of(createXrdResource(), resource),
+            ImmutableList.of(acc1, acc2),
+            balances,
+            ImmutableList.of(),
+            1L,
+            1L);
+
+    final var config =
+        new OlympiaToBabylonConverterConfig(
+            10, 10, 10, 10, 10, Decimal.unsafeFromRawBigIntRepr(BigInteger.valueOf(6000L)));
+    final var converted = OlympiaStateToBabylonGenesisConverter.toGenesisData(olympiaState, config);
+
+    // There are two balance entries: 10000 and 2001 (12k + 1 total), and the maximum supply is 6k.
+    // The balances should be scaled by (6000/12001) and one of them rounded down.
+    // The new total supply should then be 5999.
+    final var resourcesChunk = converted.chunks().get(0);
+    final var resourceBalancesChunk = (GenesisDataChunk.ResourceBalances) converted.chunks().get(1);
+
+    assertEquals(
+        Decimal.unsafeFromRawBigIntRepr(BigInteger.valueOf(5999L)),
+        ((GenesisDataChunk.Resources) resourcesChunk).value().get(0).initialSupply());
+
+    assertEquals(
+        Decimal.unsafeFromRawBigIntRepr(BigInteger.valueOf(4999L)),
+        resourceBalancesChunk.allocations().get(0).last().get(0).amount());
+
+    assertEquals(
+        Decimal.unsafeFromRawBigIntRepr(BigInteger.valueOf(1000L)),
+        resourceBalancesChunk.allocations().get(0).last().get(1).amount());
+  }
+
+  @Test
+  public void test_large_supply_exceeding_u256() {
+    final var addrBytes = new byte[27];
+    random.nextBytes(addrBytes);
+    addrBytes[0] = REAddr.REAddrType.HASHED_KEY.byteValue();
+    final var addr = REAddr.of(addrBytes);
+    final var resource =
+        new OlympiaStateIR.Resource(
+            addr, UInt256.from(18), true, Optional.empty(), "LARGE", "", "", "", "");
+    final var acc1 =
+        new OlympiaStateIR.Account(
+            HashCode.fromBytes(ECKeyPair.generateNew().getPublicKey().getCompressedBytes()));
+    final var balances =
+        ImmutableList.of(
+            new OlympiaStateIR.AccountBalance(
+                0, 1, UInt256.MAX_VALUE.toBigInt().multiply(BigInteger.TWO)));
+    final var olympiaState =
+        new OlympiaStateIR(
+            ImmutableList.of(),
+            ImmutableList.of(createXrdResource(), resource),
+            ImmutableList.of(acc1),
+            balances,
+            ImmutableList.of(),
+            1L,
+            1L);
+
+    final var config =
```

Comment:
Can we instead use the actual scaling factor we'll use? EG `2^160` and check it scales correctly?

Reply:
I've added another test that uses 2^160